### PR TITLE
docs: document log order

### PR DESCRIPTION
**Pull request by chris-olszewski**
### Description

Adds documentation for `--log-order` #3916 including a callout of our special behavior on Github Actions.

### Testing Instructions

Eyes

### CODE REVIEW

File: docs/pages/repo/docs/reference/command-line-reference/run.mdx [hunk 1]
```diff
@@ -224,6 +224,32 @@ Just a quick overview.
 - `{}` allows for a comma-separated list of "or" expressions
 - `!` at the beginning of a pattern will negate the match
 
+### `--log-order`
+
+`type: string`
+
+Set type of task output order. Defaults to "auto" to let
+turbo decide based on its own heuristics.
```
Comment (anthonyshew) [inline docs/pages/repo/docs/reference/command-line-reference/run.mdx:232]: We're mentioning "using heuristics" but I'm left wondering what _are_ those heuristics? What behaviors should I be expecting to see?

File: docs/pages/repo/docs/reference/command-line-reference/run.mdx [hunk 2]
```diff
@@ -224,6 +224,32 @@ Just a quick overview.
...
+
+| option  | description                               |
+| ------- | ----------------------------------------- |
+| auto    | turbo decides based on its own heuristics |
+| stream  | show output as soon as it is available    |
```
Comment (anthonyshew) [inline docs/pages/repo/docs/reference/command-line-reference/run.mdx:237]: Start phrase with a capital letter.

File: docs/pages/repo/docs/reference/command-line-reference/run.mdx [hunk 3]
```diff
@@ -224,6 +224,32 @@ Just a quick overview.
...
+| grouped | show output grouped by task               |
```
Comment (anthonyshew) [inline docs/pages/repo/docs/reference/command-line-reference/run.mdx:238]: Capital letter as well!

File: docs/pages/repo/docs/reference/command-line-reference/run.mdx [hunk 4]
```diff
@@ -224,6 +224,32 @@ Just a quick overview.
...
+
+**Example**
+
+```shell
+turbo run build --log-order=stream
+turbo run build --log-order=grouped
+```
+
+<Callout type="info">
+  If log order is set to auto and turbo detects that it is running on Github Actions, then turbo will
```
Comment (anthonyshew) [inline docs/pages/repo/docs/reference/command-line-reference/run.mdx:248]: turbo should be code formatted. `turbo`

File: docs/pages/repo/docs/reference/command-line-reference/run.mdx [hunk 5]
```diff
@@ -224,6 +224,32 @@ Just a quick overview.
...
+  create [grouped logs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines).
+  If this isn't desired, then the behavior can be opted out of by setting the log order to stream or grouped directly.
```
Comment (anthonyshew) [inline docs/pages/repo/docs/reference/command-line-reference/run.mdx:250]:
```suggestion
  You can opt out of this behavior by setting a log order of your own.
```

**Review by anthonyshew — CHANGES_REQUESTED** [review]

**Review by arlyon — APPROVED** [review]
No comments beyond Anthony's request so approving to unblock :)

**Review by anthonyshew — APPROVED** [review]

**Comment by anthonyshew** [comment]
Oops, sorry for being a blocker! Awesome ship, @chris-olszewski!

**Pull request merged**